<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [11]:
# Membuat dataframes dari dictionary
import pandas as pd

konfirmasi = {
    'Kota Yogyakarta': [346, 51, 18],
    'Kab. Sleman': [832, 409, 22],
    'Kab. Bantul': [602, 70, 21],
    'Kab. Kulon Progo': [117, 60, 5],
    'Kab. Gunungkidul': [212, 21, 9],
    'Non DIY': [14, 4, 0]
}

dd = {'Wilayah': [], 'Sembuh': [], 'Dirawat': [], 'Meninggal': []}
for key, values in konfirmasi.items():
    dd['Wilayah'].append(key)
    dd['Sembuh'].append(values[0])
    dd['Dirawat'].append(values[1])
    dd['Meninggal'].append(values[2])

df = pd.DataFrame(dd)
df

Wilayah  Sembuh  Dirawat  Meninggal
0   Kota Yogyakarta     346       51         18
1       Kab. Sleman     832      409         22
2       Kab. Bantul     602       70         21
3  Kab. Kulon Progo     117       60          5
4  Kab. Gunungkidul     212       21          9
5           Non DIY      14        4          0

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [12]:
# Membaca data dan memasukannya kedalam suatu variabel

dfcsv = pd.read_csv('https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd')
dfcsv

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

**Jawab**:

*   Perbedaan *Series* dengan *list* dan *dictionary* adalah pada tipe data dan dimensinya. Data *value* pada *series* harus bertipe sama, sedangkan pada *list* dan *value* pada *dictionary* dapat berbeda tipe data. Objek *series* hanya bersifat satu dimensi saja.

*   *Series* adalah objek 1 dimensi sedangkan *dataframe* adalah objek 2 dimensi. Setiap kolom pada *dataframe* merupakan objek dari *series*, dan baris terdiri dari elemen yang ada pada *series*.

In [13]:
x = pd.Series([17532.17, 19346.05, 141.58, 10970.15, 14941.34], index=['EUR', 'GBP', 'JPY', 'SGD', 'USD'])
x

EUR    17532.17
GBP    19346.05
JPY      141.58
SGD    10970.15
USD    14941.34
dtype: float64

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [14]:
print('baris pertama dari kolom price:', dfcsv.price.iloc[0])
print('baris terakhir dari kolom color:', dfcsv.color.iloc[-1])

baris pertama dari kolom price: 326
baris terakhir dari kolom color: D


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

**Jawab:**

Statistika deskriptif adalah metode-metode yang berkaitan dengan pengumpulan dan penyajian suatu gugus data sehingga memberikan informasi yang berguna **(Walpole, 1993)**. Statistik deskriptif dapat mendeskripsikan data yang sedang dianalisis, tetapi tidak boleh menarik kesimpulan apapun dari data **(Yulianto, 2012)**.

Ukuran-ukurannya dapat digolongkan menjadi dua kelompok, yaitu ukuran nilai tengah dan ukuran deviasi. Ukuran nilai tengah terdiri dari rata-rata (*mean*), median dan modus. Sedangkan ukuran deviasi terdiri dari varians, simpangan baku, koefisien variasi dan nilai jarak (*range*).

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [15]:
dfcsv.describe()

Unnamed: 0         carat  ...             y             z
count  53940.000000  53940.000000  ...  53940.000000  53940.000000
mean   26970.500000      0.797940  ...      5.734526      3.538734
std    15571.281097      0.474011  ...      1.142135      0.705699
min        1.000000      0.200000  ...      0.000000      0.000000
25%    13485.750000      0.400000  ...      4.720000      2.910000
50%    26970.500000      0.700000  ...      5.710000      3.530000
75%    40455.250000      1.040000  ...      6.540000      4.040000
max    53940.000000      5.010000  ...     58.900000     31.800000

[8 rows x 8 columns]

In [16]:
print('nilai rata-rata dari column price:', dfcsv.price.mean())
print('nilai standar deviasi dari column depth:', dfcsv.depth.std())
print('nilai maksimum dari column carat:', dfcsv.carat.max())

nilai rata-rata dari column price: 3932.799721913237
nilai standar deviasi dari column depth: 1.4326213188337733
nilai maksimum dari column carat: 5.01


**Jawab:**
*   Nilai rata-rata dari *column* `price` adalah **3932.799721913237**
*   Nilai standar deviasi dari *column* `depth` adalah **1.4326213188337733**
*   Nilai maksimum dari *column* `carat` adalah **5.01**